### PR TITLE
Pubmatic Bid Adapter: support permutive RTD

### PR DESCRIPTION
**Pull request by pm-manasi-moghe**
## Type of change

- [ ] Feature

## Description of change
Add support for pubmatic bidder in permutive module

### CODE REVIEW

**Comment by patmmccann** [comment]
I believe we should require permutive to comply with module rules instead of allowing these sort of pra
S

File: modules/pubmaticBidAdapter.js [hunk 1]
```diff
@@ -737,6 +739,44 @@ function _addFloorFromFloorModule(impObj, bid) {
...
+function _populateSegmentDataIfAvailable(fpdUserData, segmentData) {
+  // fpdUserData - data set from global and bidder level configs
+  // segmentData - data set from permutive
+  var dataMerged = false;
+
+  if ((!fpdUserData || !fpdUserData.data) && (!segmentData || segmentData.length === 0)) {
+    return;
+  }
+
+  if ((!fpdUserData || !fpdUserData.data) && segmentData.length > 0) {
```
Comment (mike-chowla) [inline modules/pubmaticBidAdapter.js:751]: If fpdUserData is set but segmentData is not, this line throws an exception.

I saw this exception testing the the example pbjs.setConfig() from the First Party Data page:
https://docs.prebid.org/features/firstPartyData.html

Screen Shot 2021-06-23 at 11 45 06 AM

**Review by mike-chowla — CHANGES_REQUESTED** [review]

**Review by mmoschovas — CHANGES_REQUESTED** [review]

**Comment by patmmccann** [comment]
See the discussion at #7001

File: modules/pubmaticBidAdapter.js (hunk 1 above)
Comment (pm-manasi-moghe) [reply modules/pubmaticBidAdapter.js:751]: This has been fixed

File: modules/pubmaticBidAdapter.js [hunk 2]
```diff
@@ -737,6 +739,44 @@ function _addFloorFromFloorModule(impObj, bid) {
   impObj.bidfloor = ((!isNaN(bidFloor) && bidFloor > 0) ? bidFloor : UNDEFINED);
 }
 
+function _populateSegmentDataIfAvailable(fpdUserData, segmentData) {
```
[the remark this answers is not in the document]
Comment (pm-manasi-moghe) [reply modules/pubmaticBidAdapter.js]: This has been taken care of in the latest commit.

**Comment by pm-manasi-moghe** [comment]
Can this PR be taken up for review?

**Comment by pm-manasi-moghe** [comment]
@jsnellbaker @mmoschovas Can this be taken up for review?

**Comment by mmoschovas** [comment]
@pm-manasi-moghe it was my understanding that there was a compliance issue here as per @patmmccann comment : https://github.com/prebid/Prebid.js/pull/7001. Have you reviewed the link posted here?

**Comment by patmmccann** [comment]
#7001 appears to be getting sorted out, @pm-manasi-moghe do you want to base your pr on that? Do you see any conflict between that and your pr? Does the approach being taken in 7001 enable support for your adapter without this pr occuring?

**Comment by pm-manasi-moghe** [comment]
@patmmccann Thanks for the updates. Let me check if the changes in #7001 can work as is for us.
The changes in permutive module in our PR may not be required post this, but need to confirm if pubmatic adapter can process the segment data without any changes at our end.

**Comment by ChrisHuie** [comment]
@pm-manasi-moghe is this pr still valid and being worked on? We are looking at older prs and trying to close out the ones that are stale :)

**Comment by mmoschovas** [comment]
@ChrisHuie if there arent any updates here, can we close this PR? This has been stale for a long time now

**Pull request closed without merging**